ros2service/ros2service/verb/echo.py

```diff
@@ -0,0 +1,236 @@
+        self.no_arr = args.no_arr
+        self.no_str = args.no_str
+        self.csv = args.csv
+        self.include_message_info = args.include_message_info
+        self.client = args.client
+        self.server = args.server
+
+        if args.service_type is None:
+            with NodeStrategy(args) as node:
+                self.srv_module = get_service_class(
+                    node, args.service_name, blocking=False, include_hidden_services=True)
+        else:
+            try:
+                self.srv_module = get_service(args.service_type)
+            except (AttributeError, ModuleNotFoundError, ValueError):
+                raise RuntimeError("The service type '%s' is invalid" % args.service_type)
+
+        if self.srv_module is None:
+            raise RuntimeError(
+                'Could not load the type for the passed service')
+
+        event_topic_name = args.service_name + "/_service_event"
+
+        with NodeStrategy(args) as node:
+            self.subscribe_and_spin(
+                node,
+                event_topic_name,
+                self.event_msg_type
+            )
+
+    def subscribe_and_spin(
+            self,
+            node: Node,
+            event_topic_name: str,
+            event_msg_type: MsgType,
+    ) -> Optional[str]:
+        """Initialize a node with a single subscription and spin."""
+
+        node.create_subscription(
+            event_msg_type,
+            event_topic_name,
+            self._subscriber_callback,
+            self.qos_profile)
+
+        rclpy.spin(node)
+
+    def _subscriber_callback(self, msg, info):
+        deserialize_msg_type = None
+        self.event_enum = msg.info.event_type
+
+        if self.client and self.server:
+            pass
+        elif self.client:
+            if self.event_enum is ServiceEventType.REQUEST_RECEIVED or \
+                    self.event_enum is ServiceEventType.RESPONSE_SENT:
+                return
+        elif self.server:
+            if self.event_enum is ServiceEventType.REQUEST_SENT or \
+                    self.event_enum is ServiceEventType.RESPONSE_RECEIVED:
+                return
+
+        if self.event_enum is ServiceEventType.REQUEST_RECEIVED or \
+                self.event_enum is ServiceEventType.REQUEST_SENT:
+            deserialize_msg_type = self.srv_module.Request
+        elif self.event_enum is ServiceEventType.RESPONSE_RECEIVED or \
+                self.event_enum is ServiceEventType.RESPONSE_SENT:
+            deserialize_msg_type = self.srv_module.Response
+        else:  # TODO remove this once event enum is correct
+            print("Returning invalid service event type")
+            return
+
+        # csv
+        if self.csv:
+            to_print = message_to_csv(
+                msg,
+                truncate_length=self.truncate_length,
+                no_arr=self.no_arr,
+                no_str=self.no_str,
+                deserialize_msg_type=deserialize_msg_type
```

Comment:
Following our discussion in https://github.com/ros2/rosidl_runtime_py/pull/17, I think we can continue to call `message_to_csv` here, deserialized the submessage here, and append the submessage to the output before printing. I think this works out because the request/response data is at the very end of the service event message.

Hopefully it's easy to do the same for the YAML format below.